### ISSUE TITLE
Landing page Search bar > When? date picker

### ISSUE THREAD

**Issue by iteles** [charterindex/application#77]
As a CI customer 
I want to be able to quickly and easily search for yachts available on the specific dates I want to go on holiday
So that I don't have to sift through hundreds of irrelevant search results.

Part of #19 

### Desktop
+ [ ] When anywhere in the 'When?' box is clicked, a date picker appears 
<img width="541" alt="screen shot 2018-04-06 at 11 37 36" src="https://user-images.githubusercontent.com/4185328/38417161-29d4c4b2-398f-11e8-836c-216cc3744454.png">

+ [ ] The current date, tomorrow and the day after are greyed out and not clickable
+ [ ] Clicking on the first date marks it as the 'From' date and the second date you click on is marked as the 'To' date
+ [ ] If a third date is clicked, the initial selection is cleared and this new date is marked as the 'From' date, with the next clicked-on date marked as the 'To' date and so on

![airbnb-datepicker-gif](https://media.giphy.com/media/nE4nL9zoAhU2wr9CsT/giphy.gif)

+ [ ] The colour bar is dragged between the 'From' and 'To' date as you select them like on Airbnb
+ [ ] Dates can be selected no further than 18 months from today's date
+ [ ] Selecting a date is not _mandatory_ (this field can be left blank in the search)
+ [ ] @Cleop & @SimonLab to determine how the date selection is saved in the initial version : clicking anywhere on the screen off the date picker? Clicking an 'Apply date selection` button? Automatically shuts date picker once a 'To' date is chosen?
+ [ ] When dates are chosen, they appears as a `pill` in the search bar in the format below, with an `X` next to them
<img src="https://user-images.githubusercontent.com/16775804/38045598-208af1fa-32b5-11e8-8a5d-0829215a85b4.png" width=150px />

+ [ ] When `X` is clicked, the selected dates disappear and the original drop-down replaces  

### Mobile
+ [ ] 'From' date & 'To' date appear as two individual fields
+ [ ] 'From' date can be selected from drop-down
+ [ ] 'To' date can be selected from drop-down

### General
+ [ ] Includes placeholder texts `When?`
+ [ ] Field & Dropdown will be styled as per designs
+ [ ] Transitions are smooth and do not refresh the page
+ [ ] Behaviour with JavaScript switched off is tested and documented in this issue for PO

**Desktop view:**
![image](https://user-images.githubusercontent.com/16775804/38044404-8f2fe1d2-32b1-11e8-8f2b-92fe5bca8f47.png)
**Mobile view:**
<img src="https://user-images.githubusercontent.com/16775804/38044430-a3748792-32b1-11e8-9527-90d6e07ebfb9.png" width=200px />

**Comment by chriswebley** [comment]
@iteles The pill showing the dates might need to include the year as well because people quite often want a yacht over Christmas and new year. Not 100% sure its necessary but possibly?

**Comment by iteles** [comment]
Agreed, thanks @chriswebley, I've added this to the acceptance criteria 👍 
Other than that, can you confirm that the issue is PO-approved please?

**Comment by SimonLab** [comment]
@Danwhy @shouston3 I think you have recently created an Elm datepicker. Do you know if your Elm package matches some of our acceptance criteria :arrow_up: , especially the "from/to" date range?

**Comment by SimonLab** [comment]
I'm going to try to integrate https://github.com/elm-community/elm-datepicker to the application and see if we can customise all the needed parts

**Comment by SimonLab** [comment]
A simple calendar can be displayed with `elm-datepicker` before trying to modify the default css/style of the calendar I'll try to integrate a fork of the plugin which allow to select a range of dates:
https://github.com/regalii/elm-datepicker

**Comment by SimonLab** [comment]
I've build a small prototype to test the underlining of a range of dates similar to:
![airbnb-datepicker-gif](https://media.giphy.com/media/nE4nL9zoAhU2wr9CsT/giphy.gif)

The code is on Ellie: https://ellie-app.com/ntsbFRD3Qa1/0

**Comment by SimonLab** [comment]
[Date Elm package]() doesn't provide comparaison function

**Comment by shouston3** [comment]
I would use [Date-Extra](http://package.elm-lang.org/packages/justinmimbs/elm-date-extra/3.0.0/Date-Extra#compare) for this

**Comment by iteles** [comment]
Mobile UI for this moved to Sprint 2.
